examples/providers/DeviceInfoProviderImpl.cpp

```diff
@@ -42,9 +43,24 @@ DeviceInfoProviderImpl & DeviceInfoProviderImpl::GetDefaultInstance()
     return sInstance;
 }
 
+// !!!!!!!!!!!!!!!!!!!!!!!! WARNING WARNING WARNING !!!!!!!!!!!!!!!!!!!!
+// WARNING: DO NOT USE THESE DEFAULT IMPLEMENTATIONS WITH DEFAULT VALUES
+// IN PRODUCTION PRODUCTS WITHOUT AUDITING THEM! See
+// `AllClustersExampleDeviceInforProviderImpl.cpp` for an example provider
+// that has constant values. Here, all providers have empty implementations
+// to force empty lists which prevent bad values from leaking into products
+// like happened before Matter 1.5. If you really are using these clusters,
+// then please re-implement the provider as needed.
+//
+// The FixedLabel, LocalizationConfigurationand and Time Format localization
+// clusters, if used, should have values that have been vetted
+// for correctness in the product !!! DO NOT USE SAMPLE DEFAULTS IN PRODUCTS.
+// !!!!!!!!!!!!!!!!!!!!!!!! WARNING WARNING WARNING !!!!!!!!!!!!!!!!!!!!
+
 DeviceInfoProvider::FixedLabelIterator * DeviceInfoProviderImpl::IterateFixedLabel(EndpointId endpoint)
 {
-    return chip::Platform::New<FixedLabelIteratorImpl>(endpoint);
+    // Will cause empty list.
+    return nullptr;
 }
 
 DeviceInfoProviderImpl::FixedLabelIteratorImpl::FixedLabelIteratorImpl(EndpointId endpoint) : mEndpoint(endpoint)
```

Comment:
Is this FixedLabelIteratorImpl thing even used?  Can't it just be removed completely?

Or is the idea to show how you would go about implementing it, in which case it might be better for `IterateFixedLabel` to actually keep returning an instance of this thing, and then the "empty list" is handled by Next() returning false...

Reply:
It is used by the FixedLabel cluster. My next PR will remove this from every sample except All Clusters App, at which point I will make it even more "lean".

Reply:
Where does FixedLabel use the concrete type?  I would expect it does not do that..  It uses `DeviceInfoProvider::FixedLabelIterator`.

Reply:
> My next PR will remove this from every sample except All Clusters App

I was looking at this as well some time ago and I noticed that as long as the platform had the FixedLabelIteratorImpl it would be "ok" since it would actually return something in the fixed label and I could not find a nice solution where it would be removed from some platforms and not others, but I guess to your point. Let us remove it and make people actively add it in if they need it, rather than having it stick around and people forget to remove it on shipping devices.

Reply:
@tennessee-google, something like https://github.com/project-chip/connectedhomeip/pull/38766? 😄

Reply:
> So the concrete type is not directly used

Right, my point is that we are not using this concrete type anywhere.  We never create instances of it.  Why not just remove all that code?

Reply:
Will do